src/result_processor.c

```diff
@@ -419,7 +419,7 @@ static int rpsortNext_innerLoop(ResultProcessor *rp, SearchResult *r) {
 
     // If there is no sorting vector, load all required fields, else, load missing fields
     if (nLoadKeys == REDISEARCH_UNINITIALIZED) {
-      if (!h->rowdata.sv) {
+      if (!h->rowdata.sv && !h->rowdata.dyn) {
```

Comment:
If you're cheap on lines of code, at least be large with the comments ;)
What does `h->rowdata.dyn` stands for? How this condition impacts the keys that the sorter will load?

Reply:
I thought my general explanation is enough in this case.
`rowdata.sv` is the list of all sortable fields in the index.
`rowdata.dyn` is the list of all loaded fields right now.
our RP is creating a new field for the query runtime, and this field is stored in the `rowdata.dyn` list.

Reply:
this change skips an optimization that doesn't check if it needs to load some fields and assumes it does, and this caused the sorter to try and load the score field for every doc. the load fails quietly, but the data is already loaded so the correctness wasn't effected

Reply:
put everything you wrote here as a comment in the code please